pennylane/templates/subroutines/qsvt.py

```diff
@@ -41,6 +40,29 @@
 from .prepselprep import PrepSelPrep
 from .qubitization import Qubitization
 
+is_jax_available = True
+is_optax_available = True
+try:
+    import jax
+except ImportError:  # pragma: no cover
+    is_jax_available = False  # pragma: no cover
+
+try:
+    import optax
+except ImportError:
+    is_otpax_available = False
```

Comment:
To avoid global variables, these could be encapsulated into the availability checks in a function or a dependencies dataclass instead of mutating module-level state.

``` python
def check_dependencies():
    try:
        import jax
        jax_available = True
    except ImportError:  # pragma: no cover
        jax_available = False

    try:
        import optax
        optax_available = True
    except ImportError:  # pragma: no cover
        optax_available = False

    return jax_available, optax_available
```

Another way to tackle this here would be by using `importlib.util.find_spec`:

``` python
def is_available(module_name) -> bool:
    return importlib.util.find_spec(module_name) is not None
```

Reply:
I tried wrapping `jax` and `optax` availability in a function closure and there would be quite some more local imports in this PR to support the feature it tries to add.
See https://github.com/PennyLaneAI/pennylane/tree/local-jax-import for details. And would like to hear what @cvqmm thinks about this.